diff --git a/empowering_customize/giscedata_polissa.py b/empowering_customize/giscedata_polissa.py
index 5bcce68..71dce8b 100644
--- a/empowering_customize/giscedata_polissa.py
+++ b/empowering_customize/giscedata_polissa.py
@@ -22,7 +22,7 @@ class GiscedataPolissa(osv.osv):
         )
     }
 
-    def send_empowering_report(self, cursor, uid, ids, context=None):
+    def send_empowering_report(self, cursor, uid, reports, context=None):
         if context is None:
             context = {}
 
@@ -33,10 +33,22 @@ class GiscedataPolissa(osv.osv):
         cl_obj = self.pool.get('empowering.customize.profile.channel.log')
         pe_send_obj = self.pool.get('poweremail.send.wizard')
         imd_obj = self.pool.get('ir.model.data')
+        attach_obj = self.pool.get('ir.attachment')
 
-        for polissa in self.browse(cursor, uid, ids, context=context):
+        for report in reports:
+            polissa = self.browse(cursor, uid, report['contract_id'],
+                context=context)
             if not polissa.empowering_profile_id:
                 continue
+
+            vals = {
+                'name': report['report_name'],
+                'datas': report['pdf'],
+                'datas_fname': report['report_name'],
+                'res_model': 'giscedata.polissa',
+                'res_id': polissa.id
+            }
+            attachment_id = attach_obj.create(cursor, uid, vals)
             now = datetime.now()
             for channel in polissa.empowering_profile_id.channels_ids:
                 measure = channel.interval_id.measure
@@ -58,6 +70,7 @@ class GiscedataPolissa(osv.osv):
                 period = context.get('period')
                 if not period:
                     period = now.strftime('%Y%m')
+                body_personal = context.get('body_personal', '')
                 channel_code = channel.channel_id.code
 
                 template_id = imd_obj.get_object_reference(
@@ -67,6 +80,7 @@ class GiscedataPolissa(osv.osv):
                 ctx = context.copy()
                 ctx.update({
                     'period': period,
+                    'body_personal': body_personal,
                     'empowering_channel': channel_code,
                     'src_rec_ids': [polissa.id],
                     'src_model': 'giscedata.polissa',
@@ -81,6 +95,8 @@ class GiscedataPolissa(osv.osv):
                     )
                 )
                 send_id = pe_send_obj.create(cursor, uid, {}, context=ctx)
+                pe_send_obj.write(cursor, uid, send_id,
+                    {'attachment_ids': [attachment_id]}, context=ctx)
                 sender = pe_send_obj.browse(cursor, uid, send_id, context=ctx)
                 sender.send_mail(context=ctx)
 
